File: skolkval/klockan/submissions/partially_accepted/gustav_n3.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
typedef vector<int> vi;

vi e {6, 2, 5, 5, 4, 5, 6, 3, 7, 6};

int energy(int time){
    assert(time >= 0 && time < 24*60*60);
    int secs = time%60;
    int mins = time/60%60;
    int hours = time/3600%24;
    return e[secs%10] + e[secs/10] +
           e[mins%10] + e[mins/10] +
           e[hours%10] + e[hours/10];
}

int interval(int a, int b){
    assert(a >= 0 && b >= a && b < 24*60*60);
    int ans = 0;
    for(int i = a; i <= b; i++) ans += energy(i);
    return ans;
}

int main(){
    int n;
    cin >> n;

    int ans = 0;
    
    for(int a = 0; a < 24*60*60; a++){
        for(int b = a; b < 24*60*60; b++){
            int s = interval(a, b);
            if(s == n) ans++;
            if(s > n) break;
        }
    }

    cout << ans << "\n";
}

File: skolkval/klockan/submissions/partially_accepted/gustav_n3.py
e = [6, 2, 5, 5, 4, 5, 6, 3, 7, 6]

def energy(p):
    seconds = p%60
    p//=60
    minutes = p%60
    p//=60
    hours = p
    return(e[seconds//10] + e[seconds%10] +
           e[minutes//10] + e[minutes%10] +
           e[hours // 10] + e[hours % 10])

n = int(input())
ans = 0
for a in range(24*60*60):
    for b in range(a, 24*60*60):
        s = 0
        for c in range(a, b+1):
            s += energy(c)
        if(s == n): ans += 1
        if(s > n): break
print(ans)
